companion.py

```diff
@@ -1070,7 +1076,8 @@ def fleetcarrier(
                 tk_response_event=tk_response_event,
                 query_time=query_time,
                 play_sound=play_sound,
-                auto_update=auto_update
+                auto_update=auto_update,
+                cmdr=monitor.cmdr
```

Comment:
This gets `cmdr` back into EDMarketConnector.py:AppWindow.capi_handle_response() ... but then you don't seem to have added anything to populate `capi_response.capi_data.request_cmdr` with it.

Reply:
Overall I'd adjust `capi_single_query()` to store a copy of `monitor.cmdr` just before the actual `.get()` of the query, and then fill in `request_cmdr` *in CAPIData` when it creates that.  Do the latter by adjusting the CAPIData `__init__()` method.

i.e. it turns out there's no need to mess with EDMCAPIRequest/Response, as it goes into the CAPIData that is held in the latter.

Reply:
I'm setting `capi_data['request_cmdr']` inside the `EDMCCAPIResponse` instance (`companion.py` line 600), so it's there for every instance, whatever the CAPI request that triggered it.

Reply:
1. That's wrong, as it shouldn't be **in the dictionary**.  It should be a property on `CAPIData` instead.
2. If we're doing it for `/fleetcarrier` there's no reason to not do it for all the CAPI queries, hence my suggestion to centralise it in `capi_single_query()`.